File: apps/core-api/lib/lotta_web/schema/tenants/feedback.ex
defmodule LottaWeb.Schema.Tenants.Feedback do
  @moduledoc false

  use Absinthe.Schema.Notation

  object :feedback do
    field :id, :id
    field :inserted_at, :datetime
    field :updated_at, :datetime
    field :topic, :string
    field :content, :string
    field :is_new, :boolean
    field :is_forwarded, :boolean
    field :is_responded, :boolean
    field :metadata, :string

    field :user, :user, resolve: Absinthe.Resolution.Helpers.dataloader(Lotta.Accounts)
  end

  input_object :create_feedback_input do
    field :topic, non_null(:string)
    field :content, non_null(:string)
    field :metadata, :string
  end
end

File: apps/webapp/src/test/fixtures/Feedback.ts
import { FeedbackModel } from 'model';

export const newFeedback: FeedbackModel = {
  __typename: 'Feedback',
  id: '6543-feed-back-1234',
  insertedAt: new Date().toISOString(),
  updatedAt: new Date().toISOString(),
  isForwarded: false,
  isResponded: false,
  isNew: true,
  topic: 'Feedback A',
  content: 'Feedback A content',
};

export const readFeedback: FeedbackModel = {
  __typename: 'Feedback',
  id: '6973-feed-back-1432',
  insertedAt: new Date().toISOString(),
  updatedAt: new Date().toISOString(),
  isForwarded: false,
  isResponded: false,
  isNew: false,
  topic: 'Feedback B',
  content: 'Feedback B content',
};

export const respondedFeedback: FeedbackModel = {
  __typename: 'Feedback',
  id: '1111-feed-back-1234',
  insertedAt: new Date().toISOString(),
  updatedAt: new Date().toISOString(),
  isForwarded: false,
  isResponded: true,
  isNew: false,
  topic: 'Feedback C',
  content: 'Feedback C content',
};

export const forwardedFeedback: FeedbackModel = {
  __typename: 'Feedback',
  id: '9999-feed-back-1234',
  insertedAt: new Date().toISOString(),
  updatedAt: new Date().toISOString(),
  isForwarded: true,
  isResponded: false,
  isNew: false,
  topic: 'Feedback D',
  content: 'Feedback D content',
};

export const feedbacks = [
  newFeedback,
  readFeedback,
  respondedFeedback,
  forwardedFeedback,
];
